### PR TITLE
Custom roles & privileges: Revise, reorder, add serverless requirements pages

### PR DESCRIPTION
**Pull request by joepeeples**
Contributes to https://github.com/elastic/security-docs-internal/issues/11. 

Also incorporates changes originally drafted in https://github.com/elastic/security-docs/pull/5763 and https://github.com/elastic/security-docs/pull/5772, but consolidated into this branch.

# Previews

## Preview serverless
Generating a preview is tricky, because a lot of content currently is hidden behind the `serverlessCustomRoles` docs [feature flag](https://github.com/elastic/docs-site/blob/main/config/variables/flags.js) until the feature is available in Production (ETA TBA). I temporarily commented out the conditionals to show the content for review purposes, and this is the latest preview build without the conditionals:

https://elastic-dot-co-docs-production-bc2c75hpv-elastic-dev.vercel.app/current/serverless/security/what-is-security-serverless

I then un-commented the conditionals to re-hide the new content behind the flag. (This is more complicated than preferred, but I'd rather get this content merged soon — even with the added complexity — so we can continue to build around it, versus the content only existing in a PR branch and getting stale).
- To preview the content that will eventually get published once the custom roles feature is available in Prod, use the link above and navigate to the respective serverless pages:
   - Elastic Security requirements
   - Elastic Defend feature privileges
   - Cases requirements
   - Entity risk scoring prerequisites
   - Network map data requirements
   - Machine learning job and rule requirements - Very minor edits; actual requirements still very unclear but can be documented further in a separate issue/PR.
   - Advanced settings
   - Endpoint response actions

## Preview ESS
These pages have minor edits for consistency with serverless counterparts:
- [Elastic Security requirements](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/sec-requirements.html)
- [Detections requirements](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/detections-permissions-section.html)
- [Elastic Defend feature privileges](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/endpoint-management-req.html)
- [Cases requirements](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/case-permissions.html)
- [Entity risk scoring requirements](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/ers-requirements.html)
- [Machine learning job and rule requirements](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/ml-requirements.html)
- [Configure network map data](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/conf-map-ui.html)
- [Advanced settings](https://security-docs_bk_5779.docs-preview.app.elstc.co/guide/en/security/master/advanced-settings.html)

# Summary of changes

## Adding content
For serverless MVP, we omitted a lot of content because custom roles and feature privileges weren't available yet. We can add that content back in now. As much as possible, I've tried to make the serverless content match ESS exactly, but there are spots where we also need to mention serverless-specific predefined roles (since serverless customers can use either predefined or custom roles now).

In some areas, I also tried not to introduce granular differences like changing `Kibana` to `Elastic Security`, because our assumptions have changed about that (the term still appears all over the serverless UI). Unifying content across serverless and ESS is a priority now.

## Reordering pages
We no longer need a separate section of requirements pages in serverless, so this PR moves those pages to locations that match ESS docs: each page is within a feature section. 

## Aligning titles
Our requirements pages currently go by lots of names: some are "prerequisites," some are "requirements," and one is even "prerequisites _**and**_ requirements"! This PR uses a single unified name for all pages (for example, "Detections requirements"), in both ESS and serverless.

## Housekeeping
This PR also moves around the serverless source files so they'll be easier to find in their respective feature folders. Doesn't affect output but improves authoring experience.

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-egjilmaf4-elastic-dev.vercel.app

* Built with commit cb69ca1005d030c16b7ba815ec55d34876169678

> Issues? Visit #next-docs in Slack

**Comment by joepeeples** [comment]
run docs-build

**Comment by joepeeples** [comment]
run docs-build

**Review by nastasha-solomon — COMMENTED** [review]
The Serverless additions and tweaks to the ESS look great! Thank you for renaming and moving the requirement files to more suitable places too. Sometimes, quickly finding the right file to change is a third of the battle 😅 

On a bit of a tangent, I believe the work you did [here](https://github.com/elastic/security-docs/pull/5779/files#diff-12ae3466ecd97aca5fa90420ce476c96136ab3112c50ca1c7084881867c451af) takes care of most, if not all, of what need to be done to address https://github.com/elastic/security-docs/issues/5301. I can follow up with Devin after you merge this PR to see if anything is missing or needs to be slightly modified in the Serverless docs.  

Thanks again for taking care of all of this and aligning both docsets. Overall, it's a major accomplishment!

**Review by jmikell821 — COMMENTED** [review]
This was a HUGE effort, thanks so much @joepeeples! Just a couple of slight nits, none critical.

File: docs/getting-started/net-map-req.asciidoc
Comment (joepeeples) [reply docs/getting-started/net-map-req.asciidoc:17]: That's how it was previously formatted, and historically we tend to use monospace for privileges (such as in the [Detections privs table](https://www.elastic.co/guide/en/security/master/detections-permissions-section.html#enable-detections-ui), "Kibana Privileges" column). I tend to agree that bold might be more appropriate for privs and features since they're UI elements, not text that the user types, but also hesitant to change this and make it less consistent with other feature privs.

File: docs/getting-started/sec-app-requirements.asciidoc
Comment (joepeeples) [reply docs/getting-started/sec-app-requirements.asciidoc]: Applied in https://github.com/elastic/security-docs/pull/5779/commits/39c3dde9cf3c570b64ad49a8eab1fb22b118ebef.

File: docs/serverless/edr-install-config/defend-feature-privs.mdx
Comment (joepeeples) [reply docs/serverless/edr-install-config/defend-feature-privs.mdx]: Applied in https://github.com/elastic/security-docs/pull/5779/commits/39c3dde9cf3c570b64ad49a8eab1fb22b118ebef.

**Review by nastasha-solomon — APPROVED** [review]

**Pull request merged**
